src/indexes.ts

```diff
@@ -673,12 +667,12 @@ class Index<T extends Record<string, any> = Record<string, any>> {
    * @returns Promise containing an EnqueuedTask
    */
   async updateSettings(settings: Settings): Promise<EnqueuedTask> {
-    const url = `indexes/${this.uid}/settings`;
-    const task = await this.httpRequest.patch(url, settings);
+    const task = (await this.httpRequest.patch({
+      relativeURL: `indexes/${this.uid}/settings`,
+      body: settings,
+    })) as EnqueuedTaskObject;
 
-    task.enqueued = new Date(task.enqueuedAt);
-
-    return task;
+    return new EnqueuedTask(task);
   }
```

Comment:
bug: previously this did not return `EnqueuedTask`, I'm not going to mark the rest, there are more